azure-functions-java-core/src/main/java/com/microsoft/azure/serverless/functions/annotation/HttpTrigger.java

```diff
@@ -11,18 +11,108 @@
+ * {@link com.microsoft.azure.serverless.functions.HttpRequestMessage}. For example:</p>
+ *
+ * <pre>
+ * {@literal @}FunctionName("hello")
+ *  public HttpResponseMessage&lt;String&gt; helloFunction(
+ *    {@literal @}HttpTrigger(name = "req", methods = {"get"}, authLevel = AuthorizationLevel.ANONYMOUS) HttpRequestMessage&lt;Optional&lt;String&gt;&gt; request) {
+ *     ....
+ *  }</pre>
+ *
+ * <p>In this code snippet you will observe that we have a function annotated with {@code @FunctionName("hello")},
+ * which indicates that this function will be available at the endpoint /api/hello. The name of the method itself, in
+ * this case {@code helloFunction} is irrelevant for all intents and purposes related to Azure Functions. Note however
+ * that the method return type is {@link com.microsoft.azure.serverless.functions.HttpResponseMessage}, and
+ * that the first argument into the function is an {@link com.microsoft.azure.serverless.functions.HttpRequestMessage}
+ * with generic type {@code Optional<String>}. This indicates that the body of the request will potentially contain
+ * a String value.</p>
+ *
+ * <p>Most important of all however is the {@code @HttpTrigger} annotation that has been applied
+ * to this argument. In this annotation you'll note that it has been given a name, as well as told what type of requests
+ * it supports (in this case, only HTTP GET requests), and that the {@link AuthorizationLevel} is anonymous, allowing
+ * access to anyone who can call the endpoint.</p>
+ *
+ * @see com.microsoft.azure.serverless.functions.HttpRequestMessage
+ * @see com.microsoft.azure.serverless.functions.HttpResponseMessage
+ * @since 1.0.0
+ */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.PARAMETER)
 public @interface HttpTrigger {
+    /**
+     * The variable name used in function code for the request or request body.
+     * @return The variable name used in function code for the request or request body.
+     */
     String name();
 
     String dataType() default "";
 
+    /**
+     * <p>Defines the route template, controlling which request URLs your function will respond to. The default value if
+     * no route is provided is the function name specified in the {@link FunctionName} annotation, applied to each
+     * Azure Function.</p>
+     *
+     * <p>By default when you create a function for an HTTP trigger, or WebHook, the function is addressable with a
+     * route of the form {@code http://<yourapp>.azurewebsites.net/api/<funcname>}. You can customize this route using
+     * this route property. For example, a route of {@code "products/{category:alpha}/{id:int}"} would mean that the
+     * function is now addressable with the following route instead of the original route:
+     * {@code http://<yourapp>.azurewebsites.net/api/products/electronics/357}, which allows the function code to support
+     * two parameters in the address: category and id.</p>
```

Comment:
You can use `@BindingName("category") String cat` and `@BindingName("id") int id` as your function method parameters to get those route values.

Reply:
Thanks - adding a sample here to explain this was on my todo list for today, but first I had to write some code! I'll do that today and then update this text.